app/controllers/support_interface/providers_controller.rb

```diff
@@ -85,7 +88,7 @@ def applications
       @filter = SupportInterface::ApplicationsFilter.new(
         params: params.merge(provider_id: @provider.id),
       )
-      @application_forms = @filter.filter_records(ApplicationForm.all)
+      @pagy, @application_forms = @filter.filter_records(ApplicationForm.all)
```

Comment:
I guess this needs consideration. `filter_records` returns a call to `pagy` implicitly. It's used in a few places. The question is, should `filter_records` implicitly return a call to `pagy` or should it return the filtered / scoped relation and in the place where `SupportInterface::ApplicationFilter` is called pagy should be used?

I don't know that I've expressed that very well but I would think that the latter is more appropriate.

What do you think?

Reply:
Oh I see. I guess it just wasn't clear from the context that `filter_records` returns a call to `pagy`. I suppose it could be made more explicit through the naming of the method (eg `paginated_filter_records` maybe), but I tend to agree with you that `pagy` should be called when the filter is used. At least, that seems to be the patterns that we've used elsewhere. 

That would require a bit of refactoring of the ApplicationFilter class though, and that is probably beyond the scope of this work. Happy for whatever is simplest to implement.

Reply:
This is a relay ticket, I picked it up from @gms-gs to add improvements, it seems even more refactoring would be ideal but I guess we have to draw the line somewhere, I have a backlog of work that supercedes this so I'll reluctantly push this on the backburner.

Thanks for reviewing